src/components/course/tests/CourseAssociatedPrograms.test.jsx

```diff
@@ -53,8 +65,11 @@ describe('<CourseAssociatedPrograms />', () => {
 
   test('renders programs with title', () => {
     render(<CourseAssociatedProgramsWithCourseContext initialState={initialState} />);
-    initialState.course.programs.forEach((program) => {
+    initialState.course.programs.forEach((program, index) => {
       expect(screen.queryByText(program.title)).toBeInTheDocument();
+      const button = screen.getByText(program.title);
+      fireEvent.click(button);
```

Comment:
nit: why `fireEvent` vs. `userEvent`? Generally, `userEvent.click` is preferred unless there's an explicit need for `fireEvent`.